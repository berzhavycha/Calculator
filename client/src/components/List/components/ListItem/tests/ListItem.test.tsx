import { render } from '@testing-library/react';
import { ListItem } from '../index';
import { vi } from 'vitest'

describe('ListItem component', () => {
    test('matches the snapshot', () => {
        const mockProps = {
            content: 'Item content',
            onItemClick: vi.fn(),
        };

        const { container } = render(
            <ListItem
                content={mockProps.content}
                onItemClick={mockProps.onItemClick}
            />
        );

        expect(container).toMatchSnapshot();
    });
})