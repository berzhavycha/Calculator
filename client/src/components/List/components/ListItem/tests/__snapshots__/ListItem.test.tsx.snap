// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ListItem component > matches the snapshot 1`] = `
<div>
  <li
    class="mb-2 p-1 pr-2 bg-white rounded-md shadow-md hover:bg-gray-100 text-xs cursor-pointer"
  >
    <p
      class="text-blue-500 font-bold text-right"
    >
      Item content
    </p>
  </li>
</div>
`;
